From 73bd1be998e9effd633fd2fc4428f8081b231fb6 Mon Sep 17 00:00:00 2001
From: Exemplar Exemplaris <exemplar@example.com>
Date: Wed, 26 Mar 2025 09:52:22 -0700
Subject: [PATCH] remove first builder

---
 subdir/builders.json | 1 -
 1 file changed, 1 deletion(-)

diff --git a/subdir/builders.json b/subdir/builders.json
index 330ecef..9acec8c 100644
--- a/subdir/builders.json
+++ b/subdir/builders.json
@@ -1,4 +1,3 @@
 [
-  "example-builder",
   "another-builder"
 ]
