diff --git a/node_modules/react-helmet/es/Helmet.js b/node_modules/react-helmet/es/Helmet.js
index 11d745a..9da3d26 100644
--- a/node_modules/react-helmet/es/Helmet.js
+++ b/node_modules/react-helmet/es/Helmet.js
@@ -237,7 +237,23 @@ var getTagsFromPropsList = function getTagsFromPropsList(tagName, primaryAttribu
                 var lowerCaseAttributeKey = attributeKey.toLowerCase();
 
                 // Special rule with link tags, since rel and href are both primary tags, rel takes priority
-                if (primaryAttributes.indexOf(lowerCaseAttributeKey) !== -1 && !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "canonical") && !(lowerCaseAttributeKey === TAG_PROPERTIES.REL && tag[lowerCaseAttributeKey].toLowerCase() === "stylesheet")) {
+                if (
+                    primaryAttributes.indexOf(lowerCaseAttributeKey) !== -1 &&
+                    !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "canonical") &&
+                    // PATCH by weiweilin:
+                    //
+                    // When rel="icon", use "rel" as the primary key.
+                    // This is needed to ensure multiple
+                    // `<link rel="icon" ... />` overrides each other properly
+                    // (see https://github.com/nfl/react-helmet/issues/279).
+                    //
+                    // A proper fix would require a fundamental redesign of how
+                    // react-helmet determines whether one component should
+                    // override another. At which point it will might be easier
+                    // to build a new libaray.
+                    !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "icon") &&
+                    !(lowerCaseAttributeKey === TAG_PROPERTIES.REL && tag[lowerCaseAttributeKey].toLowerCase() === "stylesheet")
+                ) {
                     primaryAttributeKey = lowerCaseAttributeKey;
                 }
                 // Special case for innerHTML which doesn't work lowercased
diff --git a/node_modules/react-helmet/lib/Helmet.js b/node_modules/react-helmet/lib/Helmet.js
index d7f564c..59fc037 100644
--- a/node_modules/react-helmet/lib/Helmet.js
+++ b/node_modules/react-helmet/lib/Helmet.js
@@ -243,7 +243,23 @@ var getTagsFromPropsList = function getTagsFromPropsList(tagName, primaryAttribu
                 var lowerCaseAttributeKey = attributeKey.toLowerCase();
 
                 // Special rule with link tags, since rel and href are both primary tags, rel takes priority
-                if (primaryAttributes.indexOf(lowerCaseAttributeKey) !== -1 && !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "canonical") && !(lowerCaseAttributeKey === TAG_PROPERTIES.REL && tag[lowerCaseAttributeKey].toLowerCase() === "stylesheet")) {
+                if (
+                    primaryAttributes.indexOf(lowerCaseAttributeKey) !== -1 &&
+                    !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "canonical") &&
+                    // PATCH by weiweilin:
+                    //
+                    // When rel="icon", use "rel" as the primary key.
+                    // This is needed to ensure multiple
+                    // `<link rel="icon" ... />` overrides each other properly
+                    // (see https://github.com/nfl/react-helmet/issues/279).
+                    //
+                    // A proper fix would require a fundamental redesign of how
+                    // react-helmet determines whether one component should
+                    // override another. At which point it will might be easier
+                    // to build a new libaray.
+                    !(primaryAttributeKey === TAG_PROPERTIES.REL && tag[primaryAttributeKey].toLowerCase() === "icon") &&
+                    !(lowerCaseAttributeKey === TAG_PROPERTIES.REL && tag[lowerCaseAttributeKey].toLowerCase() === "stylesheet")
+                ) {
                     primaryAttributeKey = lowerCaseAttributeKey;
                 }
                 // Special case for innerHTML which doesn't work lowercased
