diff --git a/node_modules/ts-proto/build/generate-services.js b/node_modules/ts-proto/build/generate-services.js
index 8117f7f..e3d6fe6 100644
--- a/node_modules/ts-proto/build/generate-services.js
+++ b/node_modules/ts-proto/build/generate-services.js
@@ -158,6 +158,7 @@ function generateServiceClientImpl(ctx, fileDesc, serviceDesc) {
     chunks.push((0, ts_poet_1.code) `export class ${name}ClientImpl${t} implements ${(0, ts_poet_1.def)(i)} {`);
     // Create the constructor(rpc: Rpc)
     const rpcType = options.context ? "Rpc<Context>" : "Rpc";
+    chunks.push((0, ts_poet_1.code) `static readonly DEFAULT_SERVICE = ${serviceNameConst};`);
     chunks.push((0, ts_poet_1.code) `private readonly rpc: ${rpcType};`);
     chunks.push((0, ts_poet_1.code) `private readonly service: string;`);
     chunks.push((0, ts_poet_1.code) `constructor(rpc: ${rpcType}, opts?: {service?: string}) {`);
