diff --git a/node_modules/@types/gapi.auth2/index.d.ts b/node_modules/@types/gapi.auth2/index.d.ts
index 7ec6781..6ba8459 100644
--- a/node_modules/@types/gapi.auth2/index.d.ts
+++ b/node_modules/@types/gapi.auth2/index.d.ts
@@ -205,13 +205,13 @@ declare namespace gapi.auth2 {
    * Reference: https://developers.google.com/api-client-library/javascript/reference/referencedocs#gapiauth2authresponse
    */
   interface AuthResponse {
-    access_token: string;
-    id_token: string;
-    login_hint: string;
-    scope: string;
-    expires_in: number;
-    first_issued_at: number;
-    expires_at: number;
+    access_token?: string;
+    id_token?: string;
+    login_hint?: string;
+    scope?: string;
+    expires_in?: number;
+    first_issued_at?: number;
+    expires_at?: number;
   }
 
   /**
@@ -272,7 +272,7 @@ declare namespace gapi.auth2 {
     /**
      * Get the user's basic profile information.
      */
-    getBasicProfile(): BasicProfile;
+    getBasicProfile(): BasicProfile | undefined;
 
     /**
      * Get the response object from the user's auth session.
