diff --git a/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js b/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
index 158b3f0..581d4ce 100644
--- a/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
+++ b/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
@@ -197,7 +197,10 @@ module.exports = async (compilation, config) => {
   } = compilation.options.output;
   const fileDetails = Array.from(filteredAssets).map(asset => {
     return {
-      file: resolveWebpackURL(publicPath, asset.name),
+      // TODO(weiweilin): remove the patch once
+      // https://github.com/GoogleChrome/workbox/issues/2856
+      // is fixed.
+      file: resolveWebpackURL(publicPath, asset.name.split(/[/\\]/).map(encodeURIComponent).join('/')),
       hash: getAssetHash(asset),
       size: asset.source.size() || 0
     };
