diff --git a/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js b/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
index 7beec5b..72c6c0a 100644
--- a/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
+++ b/node_modules/workbox-webpack-plugin/build/lib/get-manifest-entries-from-compilation.js
@@ -163,7 +163,10 @@ async function getManifestEntriesFromCompilation(compilation, config) {
     const { publicPath } = compilation.options.output;
     const fileDetails = Array.from(filteredAssets).map((asset) => {
         return {
-            file: (0, resolve_webpack_url_1.resolveWebpackURL)(publicPath, asset.name),
+            // TODO(weiweilin): remove the patch once
+            // https://github.com/GoogleChrome/workbox/issues/2856
+            // is fixed.
+            file: (0, resolve_webpack_url_1.resolveWebpackURL)(publicPath, asset.name.split(/[/\\]/).map(encodeURIComponent).join('/')),
             hash: (0, get_asset_hash_1.getAssetHash)(asset),
             size: asset.source.size() || 0,
         };
